library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -480,7 +480,11 @@ public void updateStatus(int status) {
     }
 
     private boolean isClientReadyToDownload(DownloadBatch downloadBatch) {
-        return downloadClientReadyChecker.isAllowedToDownload(downloadBatch);
+        return downloadClientReadyChecker.isAllowedToDownload(marshallToDownload(downloadBatch));
+    }
+
+    private Download marshallToDownload(DownloadBatch downloadBatch) {
+        return new Download(downloadBatch.getBatchId(), downloadBatch.getCurrentSize(), downloadBatch.getTotalSize());
```

Comment:
nice idea. 

This is the public external representation that clients can touch and mould to their hearts content. It free's our other models up to be touched and moulded by us without caring about *puny clients*.

![](http://www.quickmeme.com/img/43/43dcb3a680f3ecd72ed9350d48868f74e76e12c6fffe596ba6aa30b0bcaaacdf.jpg)